connector-webhook/src/main/java/com/redhat/cloud/notifications/connector/webhook/WebhookRouteBuilder.java

```diff
@@ -0,0 +1,75 @@
+import com.redhat.cloud.notifications.connector.webhook.authentication.BasicAuthenticationProcessor;
+import com.redhat.cloud.notifications.connector.webhook.authentication.BearerTokenAuthenticationProcessor;
+import com.redhat.cloud.notifications.connector.webhook.authentication.InsightsTokenAuthenticationProcessor;
+import org.apache.camel.builder.endpoint.dsl.HttpEndpointBuilderFactory;
+import org.apache.camel.component.http.HttpComponent;
+import org.apache.http.conn.ssl.NoopHostnameVerifier;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import static com.redhat.cloud.notifications.connector.ConnectorToEngineRouteBuilder.SUCCESS;
+import static com.redhat.cloud.notifications.connector.ExchangeProperty.*;
+import static com.redhat.cloud.notifications.connector.webhook.ExchangeProperty.BASIC_AUTH_USERNAME;
+import static com.redhat.cloud.notifications.connector.webhook.ExchangeProperty.BEARER_TOKEN;
+import static com.redhat.cloud.notifications.connector.webhook.ExchangeProperty.INSIGHT_TOKEN_HEADER;
+import static com.redhat.cloud.notifications.connector.webhook.ExchangeProperty.TARGET_URL_NO_SCHEME;
+import static com.redhat.cloud.notifications.connector.webhook.ExchangeProperty.TRUST_ALL;
+import static com.redhat.cloud.notifications.connector.webhook.SslTrustAllManager.getSslContextParameters;
+import static org.apache.camel.LoggingLevel.INFO;
+
+
+@ApplicationScoped
+public class WebhookRouteBuilder extends EngineToConnectorRouteBuilder {
+
+    @Inject
+    ConnectorConfig webhookConnectorConfig;
+
+    @Override
+    public void configureRoute() {
+
+        configureHttpsComponent();
+
+        from(direct(ENGINE_TO_CONNECTOR))
+            .routeId(webhookConnectorConfig.getConnectorName())
+            .choice()
+                .when(exchangeProperty(INSIGHT_TOKEN_HEADER))
+                    .process(new InsightsTokenAuthenticationProcessor())
+                .endChoice()
+                .when(exchangeProperty(BASIC_AUTH_USERNAME))
+                    .process(new BasicAuthenticationProcessor())
+                .endChoice()
+                .when(exchangeProperty(BEARER_TOKEN))
+                    .process(new BearerTokenAuthenticationProcessor())
+                .endChoice()
+            .end()
+            // SSL certificates may or may not be verified depending on the integration settings.
+            .choice()
+                .when(exchangeProperty(TRUST_ALL))
+                    .toD(buildUnsecureSslEndpoint(), webhookConnectorConfig.getEndpointCacheMaxSize())
+                .endChoice()
+                .otherwise()
+                    .toD("${exchangeProperty." + TARGET_URL + "}", webhookConnectorConfig.getEndpointCacheMaxSize())
+            .end()
+            .log(INFO, getClass().getName(), "Delivered event ${exchangeProperty." + ID + "} " +
+                "(orgId ${exchangeProperty." + ORG_ID + "}) " +
+                "to ${exchangeProperty." + TARGET_URL + "}")
```

Comment:
Could we log something similar to `teams` ? 

https://github.com/RedHatInsights/notifications-backend/blob/e028d207dad61481461e510649275318bc8925eb/connector-microsoft-teams/src/main/java/com/redhat/cloud/notifications/connector/microsoft/teams/TeamsRouteBuilder.java#L30

It will be useful to know the endpoint type too (if possible) since webhook is used by Ansible and regular-webhooks, unless we are planning into having a specific connector for Ansible (then i would expect a similar log in there :grin: )

Reply:
Good point ;)